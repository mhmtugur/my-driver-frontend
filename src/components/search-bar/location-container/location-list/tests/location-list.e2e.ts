import { E2EElement, E2EPage, newE2EPage } from '@stencil/core/testing';

describe('location list', () => {
  let page: E2EPage;
  let element: E2EElement;
  const locations = [
    { label: 'Berlin', address: 'Berlin Airport' },
    { label: 'Berlin', address: 'City Centre' },
    { label: 'Munich', address: 'Munich Airport' },
  ]
  beforeEach(async () => {
    page = await newE2EPage({ html: '<location-list></location-list>' });
    element = await page.find('location-list');
  });

  it('should render', async () => {
    element.setProperty('locations', locations);
    await page.waitForChanges();
    expect(element.innerHTML).toMatchSnapshot();
  });

  it('should render all locations', async () => {
    element.setProperty('locations', locations);
    await page.waitForChanges();
    const listItems = await page.findAll('.list-item');
    await page.waitForChanges();
    expect(listItems.length).toBe(locations.length);
  });

  it('should not crash if there are no locations', async () => {
    element.setProperty('locations', []);
    await page.waitForChanges();
    expect(element.innerHTML).toMatchSnapshot();
  });
})