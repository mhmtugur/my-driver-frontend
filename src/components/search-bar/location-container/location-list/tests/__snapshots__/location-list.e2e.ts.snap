// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`location list should not crash if there are no locations 1`] = `"<div></div>"`;

exports[`location list should render 1`] = `"<div><location-list-item class=\\"hydrated\\"><div class=\\"list-item\\"><i class=\\"material-icons location-icon\\">location_on</i><div><span class=\\"label\\">Berlin</span><span class=\\"address truncate\\">Berlin Airport</span></div></div></location-list-item><location-list-item class=\\"hydrated\\"><div class=\\"list-item\\"><i class=\\"material-icons location-icon\\">location_on</i><div><span class=\\"label\\">Berlin</span><span class=\\"address truncate\\">City Centre</span></div></div></location-list-item><location-list-item class=\\"hydrated\\"><div class=\\"list-item\\"><i class=\\"material-icons location-icon\\">location_on</i><div><span class=\\"label\\">Munich</span><span class=\\"address truncate\\">Munich Airport</span></div></div></location-list-item></div>"`;
